### DISCUSSION TITLE
Start with subset of the available columns

### DISCUSSION THREAD

**Discussion by clarkmcc** [lineupjs/lineupjs#622] [Q&A]
I have a dataset that has hundreds of columns, I'd like to see those columns in the sidebar viewer like they appear today, but I'd like to start lineup without all of the columns showing up in the table viewer. Is this possible? I'm using the React wrapper if that makes a difference.

**Comment by sgratzl** [comment]
take a look at the advanced usage example here: https://github.com/lineupjs/lineupjsx. In this scenario a custom ranking was defined with a subset of columns and a predefined ordering. Node that in this example also the `defaultRanking` attribute was set thus you see two rankings.

**Reply by clarkmcc** [reply]
I tried this originally but couldn't get it to work. First of all, I couldn't figure out how to define columns that were hidden by default but could be enabled through the sidebar. Secondly, I wanted to still take advantage of the fact that if I don't specify columns, lineup automatically makes them available and determines the column's type which didn't seem available when I manually specified the columns.

**Reply by sgratzl** [reply]
can you provide an example e.g. codesandbox to better understand what you try to achieve?
